tests/NumberGeneratorTest.java

```diff
@@ -7,23 +7,39 @@
 public class NumberGeneratorTest {
 
     @Test
-    public void generateNumberReturnsNewArrayListWithFourDigits(){
+    public void generateFourDigitRandomNumberReturnsNewArrayListWithFourDigits(){
         NumberGenerator numberGenerator = new NumberGenerator();
 
         int expectedResult = 4;
 
-        int actualResult = numberGenerator.generateNumber().size();
+        int actualResult = numberGenerator.generateFourDigitRandomNumber().size();
 
         assertEquals(expectedResult, actualResult);
     }
 
-//    TODO: add this test in later:
+    @Test
+    public void generateFourDigitRandomNumberReturnsNewArrayListWithFourDigitsBetweenAnd1And9(){
+        NumberGenerator numberGenerator = new NumberGenerator();
+
+        ArrayList<Integer> randomDigits = numberGenerator.generateFourDigitRandomNumber();
+
+        assertTrue(randomDigits.contains(1) || randomDigits.contains(2) || randomDigits.contains(3) || randomDigits.contains(4) ||
+                randomDigits.contains(5) || randomDigits.contains(6) && randomDigits.contains(7) || randomDigits.contains(8) || randomDigits.contains(9)
```

Comment:
Might be a stray && here, though I'm not sure the test name reflects what you're asserting.
I think the test name is a good condition to test, but you're actually just asserting (after fixing the &&) that randomDigits contains a 1,2,3,...,9 somewhere in it. 
I could be wrong, but, your array could be [9,10,0,534] and this looks like it would still pass because it still contains a 9.

Reply:
i found and eliminated the stray &&

not sure how to fix the assertion problem, but i see what you mean.

Reply:
what if i said it the other way around, that the array list does not contain anything other that the 9 digits its allowed?